////////////////////////////////////////////////////////////////////////////////
/// @file         ProtocolMessage.proto
///
/// @author       Michael Catanzaro <michael.catanzaro@mst.edu>
///
/// @description  A ProtocolMessage encapsulates a ModuleMessage.
///
/// These source code files were created at Missouri University of Science and
/// Technology, and are intended for use in teaching or research. They may be
/// freely copied, modified, and redistributed as long as modified versions are
/// clearly marked as such and this notice is not removed. Neither the authors
/// nor Missouri S&T make any warranty, express or implied, nor assume any legal
/// responsibility for the accuracy, completeness, or usefulness of these files
/// or any information distributed with these files.
///
/// Suggested modifications or questions about these files can be directed to
/// Dr. Bruce McMillin, Department of Computer Science, Missouri University of
/// Science and Technology, Rolla, MO 65409 <ff@mst.edu>.
////////////////////////////////////////////////////////////////////////////////

import "ModuleMessage.proto";

package freedm.broker;

message ProtocolMessage
{
    enum Status
    {
        CREATED = 1;
        ACCEPTED = 2;
        BAD_REQUEST = 3;
        MESSAGE = 4;
    }

    required uint32 sequence_num = 3;
    optional string expire_time = 4;
    required Status status = 5;
    optional int32 kill = 6;
    optional fixed64 hash = 7;

    optional ModuleMessage module_message = 8;
}

message ProtocolMessageWindow
{
    required string source_uuid = 1;
    required string send_time = 2;
    repeated ProtocolMessage messages = 3;
}
